src/main/java/cubicchunks/client/ClientCubeCache.java

```diff
@@ -23,99 +23,112 @@
  */
 package cubicchunks.client;
 
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
 import cubicchunks.util.CubeCoords;
-import cubicchunks.world.ICubeCache;
+import cubicchunks.util.ReflectionUtil;
 import cubicchunks.world.ICubicWorldClient;
 import cubicchunks.world.column.BlankColumn;
 import cubicchunks.world.column.Column;
 import cubicchunks.world.cube.BlankCube;
 import cubicchunks.world.cube.Cube;
+import cubicchunks.world.provider.ICubeCache;
 import net.minecraft.client.multiplayer.ChunkProviderClient;
 import net.minecraft.util.math.ChunkPos;
 import net.minecraft.world.World;
 
+//TODO: break off ICubeCache
 public class ClientCubeCache extends ChunkProviderClient implements ICubeCache {
 
 	private ICubicWorldClient world;
-	private BlankColumn blankColumn;
 	private Cube blankCube;
+	private Map<CubeCoords, Cube> cubemap = new HashMap<>();
 
 	public ClientCubeCache(ICubicWorldClient world) {
 		super((World) world);
 
 		this.world = world;
-		this.blankColumn = new BlankColumn(world, 0, 0);
-		this.blankCube = new BlankCube(world, blankColumn);
+		ReflectionUtil.setFieldValueSrg(this, "field_73238_a", new BlankColumn(this, world, 0, 0));
+		this.blankCube = new BlankCube((Column)blankChunk);
 	}
 
 	@Override
-	public Column loadChunk(int cubeX, int cubeZ) {
+	@Nullable
+    public Column getLoadedChunk(int x, int z)
```

Comment:
int cubeX, int cubeZ are preferable because they tell you what kind of coordinates are expected, instead of just x, z